gtk*: look for config files in /etc/

SuperSandro2000 opened the pull request:
###### Description of changes

This is required for gtk to read the config file from ``/etc/gtk-X.0/settings.ini``. Otherwise it would look into its nix store entry.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

r-ryantm committed:
ocamlPackages.rresult: 0.6.0 -> 0.7.0

r-ryantm committed:
ocamlPackages.ocaml_sqlite3: 5.0.2 -> 5.1.0

r-ryantm committed:
python310Packages.flufl_lock: 7.1 -> 7.1.1

Et7f3 committed:
ocamlPackages.ocaml: Migrate to integrated assembler for Darwin

This is linked to https://github.com/ocaml/ocaml/pull/10176
A similar issue appeared with clang in cctools (Darwin):
```
as: this system assembler is deprecated. Please migrate to the clang integrated assembler (`as -q').
```
https://developer.apple.com/documentation/xcode-release-notes/xcode-12-release-notes#Deprecations

jordanisaacs committed:
nqptp: init at unstable-2022-09-12

Corresponds to v1.1-dev-168-g3444047

marsam committed:
opam: 2.1.3 -> 2.1.4

https://github.com/ocaml/opam/releases/tag/2.1.4

amarshall committed:
arrow-cpp: Use minimal aws-sdk-cpp, reducing closure 1.0G->417M

evils committed:
tuptime: 5.2.1 -> 5.2.2

repo owner changed account name
  this was announced in the changelog

renamed -x option, which is currently used in the NixOS module
  this commit should be followed immediately by a module update

evils committed:
nixos/tuptime: 5.2.2 renamed option -x to -q

toelke committed:
gitkraken: fix broken interpreter after recent update

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

Stunkymonkey committed:
nix-generate-from-cpan: deprecate phases

srounce committed:
oversteer: init at 0.7.2

Add Oversteer - Steering Wheel Manager for Linux
at version [0.7.2](https://github.com/berarma/oversteer/releases/tag/0.7.2).

SuperSandro2000 committed:
gtk2: look for config files in /etc/

SuperSandro2000 committed:
gtk3: look for config files in /etc/

SuperSandro2000 committed:
gtk4: look for config files in /etc/

winterqt commented:
This may be obvious to GTK folks, but why is this change something we want? (Also: there are some changes here that probably can be split into different commits, or removed entirely, like the version interpolation.)

jtojnar commented on pkgs/development/libraries/gtk/2.x.nix:
```diff
@@ -67,6 +74,7 @@ stdenv.mkDerivation rec {
   '' else null;
 
   configureFlags = [
+    "--sysconfdir=/etc"
```
Does `installFlags = [ "sysconfdir=${placeholder "out"}/etc" ];` as mentioned in https://github.com/NixOS/nixpkgs/issues/65718 not work?

laalsaas replied:
We don't want the sysconfdir to be in `$out`, since that would mean that you still can't manage gtk config declaratively. (which this PR tries to change)

jtojnar replied:
It would not be `installFlags` would not affect what is hardcoded in the code, only the installation paths.

SuperSandro2000 replied:
but we want to change what is hardcoded in the code.

jtojnar replied:
That is what `configureFlags` are for.

Artturin replied:
> That is what `configureFlags` are for.

configureFlags is used here

jtojnar replied:
Right but `installFlags` could be used instead of the patch.

jtojnar commented on pkgs/development/libraries/gtk/patches/3.0-sysconfdir_install.patch:
```diff
@@ -0,0 +1,45 @@
+   [ 'multipress', files( [ immodule_srcdir + 'gtkimcontextmultipress.c', immodule_srcdir + 'immultipress.c' ]), true,
+                     ['-DMULTIPRESS_LOCALEDIR="@0@"'.format(gtk_mplocaledir),
+-                     '-DMULTIPRESS_CONFDIR="@0@/gtk-3.0"'.format(gtk_sysconfdir)] ],
++                     '-DMULTIPRESS_CONFDIR="@0@/gtk-3.0"'.format(gtk_sysconfdir_install)] ],
```
Does GTK handle the absence of the config file gracefully? Otherwise the input method will be broken without `environment.etc."gtk-3.0/im-multipress.conf".source = "${pkgs.gtk3}/etc/gtk-3.0/im-multipress.conf";`. (Though I am not sure if the input method used by anyone.)

SuperSandro2000 replied:
The path of this file gets hardcoded in the shared object and gtk tries to install the config to that path and read from it. For this single file nothing changed and you still can't easily change it.

jtojnar replied:
But the file will not exist in `/etc` unless user explicitly creates it, which most people won’t, thus breaking the input method for them. And I am not really convinced it makes sense to to edit the input method description in the first place.

SuperSandro2000 replied:
It is not installing into ``/etc`` but ``$out/etc``.

```
 ➜ strings result/lib/gtk-3.0/3.0.0/immodules/im-multipress.so | rg im-multipress.conf
/nix/store/nbigklvq9nimb9azglq76zyhkwlyhdik-gtk+3-3.24.35/etc/gtk-3.0/im-multipress.conf
```

jtojnar replied:
Sorry, you are right, I misread it.

laalsaas commented:
> This may be obvious to GTK folks, but why is this change something we want? 
To manage system-wide gtk configuration declaratively by `environment.etc`. Potentialy it would be nice to write a module for this in the Future

> And was this tested at all, even if on master?
This was tested on the nixos-unstable branch and then rebased onto master. Source: I was sitting next to the author when he did.

lilyinstarlight commented:
> To manage system-wide gtk configuration declaratively by `environment.etc`. Potentialy it would be nice to write a module for this in the Future

If this helps declarative config then great I'd love that, but I don't know GTK internals very well and I'm not entirely clear on what this lets you configure that you couldn't before?

(I already manage a few things like GTK theme settings in `/etc/gtk-3.0/settings.ini`, for example)

SuperSamus committed:
bottles: 2022.12.14.1 -> 50.2

r-ryantm committed:
python310Packages.pyro-ppl: 1.8.3 -> 1.8.4

fabaff committed:
python310Packages.openerz-api: add changelog to meta

fabaff committed:
python310Packages.openerz-api: 0.1.0 -> 0.2.0

Diff: https://github.com/misialq/openerz-api/compare/refs/tags/v0.1.0...v0.2.0

Changelog: https://github.com/misialq/openerz-api/releases/tag/v0.2.0

SuperSandro2000 committed:
ArchiSteamFarm: 5.4.0.3 -> 5.4.1.11

Cynerd committed:
nixos/init-script: fix missing nixos config

The d3528cdc3dbe82a0551707fe869aec02bba72956 introduced error:
  error: attribute 'nixos' missing

The correct config is clearly system.nixos thus just a simple swap and
just missed in testing.

sternenseemann committed:
lib.teams.llvm: create

As a first stop towards getting a bit more organized for #171047, add a
maintainer team and add myself and John Ericson as new members. Michael
Raskin asked to be removed.

A second step could be creating a github team additionally.

dotlambda committed:
python310Packages.paperwork-backend: use Levenshtein

srounce committed:
maintainers: add srounce

SuperSandro2000 commented:
> This may be obvious to GTK folks, but why is this change something we want?

Added the motivation to the PR body. Forgot that first. Also @laalsaas already answered it, too.

> removed entirely, like the version interpolation.

Thats the only one. To nativeBuildInputs was autoreconfHook added because autoconf triggered it automatically and I didn't want to add every tool individually. 

> And was this tested at all, even if on master?

It was tested with one gtk application for each version and by disabling the bell on errors. With this PR we have some silence when pressing backspace in an empty input.

> I'm not entirely clear on what this lets you configure that you couldn't before?

Before the config file was read from the nix store path and /etc/gtk* was never read.



> (I already manage a few things like GTK theme settings in `/etc/gtk-3.0/settings.ini`, for example)

That didn't work at least for the bell on error setting.

wegank committed:
python311Packages.proto-plus: 1.22.1 -> 1.22.2

figsoda committed:
ruff: 0.0.227 -> 0.0.228

Diff: https://github.com/charliermarsh/ruff/compare/v0.0.227...v0.0.228

Changelog: https://github.com/charliermarsh/ruff/releases/tag/v0.0.228

dotlambda committed:
paperwork: use Levenshtein

r-ryantm committed:
spicedb: 1.15.0 -> 1.16.1

r-ryantm committed:
treesheets: unstable-2023-01-04 -> unstable-2023-01-20

urandom2 committed:
signal-desktop-beta: 6.2.0-beta.2 -> 6.3.0-beta.1

It appears that signal forked ringrtc within this release, so until they
merge their changes upstream, or the stable release follows suit, we
have another delta we need to inject and track.

r-ryantm committed:
dprint: 0.34.1 -> 0.34.4

ncfavier committed:
nixos: fix backticks in Markdown descriptions

r-ryantm committed:
python310Packages.oslo-db: 12.3.0 -> 12.3.1

r-ryantm committed:
youtube-music: 1.18.0 -> 1.19.0

fabaff committed:
python310Packages.hdate: add changelog to meta

fabaff committed:
python310Packages.hdate: update postPatch

Mic92 committed:
python3.pkgs.matrix-nio: 0.20.0 -> 0.20.1

Diff: https://github.com/poljar/matrix-nio/compare/0.20.0...0.20.1

ncfavier committed:
doc/manpage-urls.json: add mpd

fabaff committed:
python310Packages.pytelegrambotapi: switch to pytestCheckHook

- add optional-dependencies
- normalize pname
- add changelog to meta

fabaff committed:
python310Packages.pytelegrambotapi: 4.8.0 -> 4.9.0

Diff: https://github.com/eternnoir/pyTelegramBotAPI/compare/refs/tags/4.8.0...4.9.0

Changelog: https://github.com/eternnoir/pyTelegramBotAPI/releases/tag/4.9.0

fabaff committed:
python310Packages.pytest-testmon: add changelog to meta

fabaff committed:
python310Packages.pytest-testmon: 1.4.3 -> 1.4.5

Diff: https://github.com/tarpas/pytest-testmon/compare/refs/tags/v1.4.3...v1.4.5

Changelog: https://github.com/tarpas/pytest-testmon/releases/tag/v1.4.5

fabaff committed:
python310Packages.pytest-isort: add changelog to meta

fabaff committed:
python310Packages.pytest-isort: 3.0.0 -> 3.1.0

Diff: https://github.com/stephrdev/pytest-isort/compare/refs/tags/3.0.0...3.1.0

Changelog: https://github.com/stephrdev/pytest-isort/blob/3.1.0/CHANGELOG.rst

symphorien committed:
paperwork: fix build

gensymb.sty used to be in the was package, but is not a standalone
package

fabaff committed:
python310Packages.yfinance: add missing inputs

fabaff committed:
python310Packages.yfinance: 0.2.3 -> 0.2.4

Diff: https://github.com/ranaroussi/yfinance/compare/refs/tags/0.2.3...0.2.4

Changelog: https://github.com/ranaroussi/yfinance/blob/0.2.4/CHANGELOG.rst

GTrunSec committed:
nixos/nomad: add LoadCredential option

fabaff committed:
python310Packages.frozendict: disable test on Python 3.11

fabaff committed:
python311Packages.merkletools: remove pysha3

r-ryantm committed:
esbuild: 0.16.17 -> 0.17.3

figsoda committed:
treewide: remove empty go vendor hashes

ncfavier committed:
doc/manpage-urls.json: add gnunet.conf

symphorien committed:
Merge pull request #211906 from Mic92/matrix-nio

python3.pkgs.matrix-nio: 0.20.0 -> 0.20.1

r-ryantm committed:
gotrue-supabase: 2.41.0 -> 2.41.4

r-ryantm committed:
jbang: 0.97.0 -> 0.102.0

SuperSandro2000 committed:
Merge pull request #165105 from r-ryantm/auto-update/ocaml4.13.1-sqlite3

SuperSandro2000 committed:
Merge pull request #208957 from marsam/update-opam

SuperSandro2000 committed:
Merge pull request #203867 from jordanisaacs/nqptp-init

zimbatm committed:
Merge pull request #211423 from srounce/add-oversteer-gui

oversteer: init at 0.7.2

fabaff committed:
Merge pull request #211913 from fabaff/pytest-isort-fix

python310Packages.pytest-isort: 3.0.0 -> 3.1.0

fabaff committed:
Merge pull request #211912 from fabaff/pytest-testmon-fix

python310Packages.pytest-testmon: 1.4.3 -> 1.4.5

SuperSandro2000 committed:
Merge pull request #197367 from Et7f3/use-integrated-as-darwin

fabaff committed:
Merge pull request #211091 from fabaff/openerz-api-bump

python310Packages.openerz-api: 0.1.0 -> 0.2.0

SuperSandro2000 committed:
Merge pull request #211101 from SuperSamus/bottles-update

SuperSandro2000 committed:
Merge pull request #211889 from urandom2/signal-desktop0

romildo committed:
vimix-icon-theme: 2021-11-09 -> 2023-01-18 (#211488)

SuperSandro2000 committed:
Merge pull request #211347 from Cynerd/fix-init-script

fabaff committed:
Merge pull request #211935 from fabaff/yfinance-fix

python310Packages.yfinance: 0.2.3 -> 0.2.4

figsoda committed:
Merge pull request #211963 from r-ryantm/auto-update/gotrue-supabase

gotrue-supabase: 2.41.0 -> 2.41.4

SuperSandro2000 committed:
Merge pull request #165097 from r-ryantm/auto-update/ocaml4.13.1-rresult

urandom2 committed:
ivpn{,-service}: init at 3.10.0 (#211888)

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

onny committed:
Merge pull request #211909 from fabaff/pytelegrambotapi-fix

python310Packages.pytelegrambotapi: 4.8.0 -> 4.9.0

onny committed:
Merge pull request #211905 from fabaff/hdate-fix

python310Packages.hdate: update postPatch

figsoda committed:
Merge pull request #211823 from figsoda/ruff

ruff: 0.0.227 -> 0.0.228

onny committed:
Merge pull request #211903 from r-ryantm/auto-update/youtube-music

youtube-music: 1.18.0 -> 1.19.0

onny committed:
Merge pull request #211900 from r-ryantm/auto-update/python310Packages.oslo-db

python310Packages.oslo-db: 12.3.0 -> 12.3.1

onny committed:
Merge pull request #211893 from r-ryantm/auto-update/dprint

dprint: 0.34.1 -> 0.34.4

SuperSandro2000 committed:
Merge pull request #209844 from toelke/update-gitkraken

fix https://github.com/NixOS/nixpkgs/pull/206240

dotlambda committed:
Merge pull request #211641 from dotlambda/paperwork-backend-fix

paperwork: use Levenshtein

SuperSandro2000 committed:
Merge pull request #209227 from amarshall/arrow-cpp-shrink-closure

cyntheticfox committed:
pass-secret-service: Add meta.mainProgram

While this package had its name updated to reflect nix package naming
style, the binary's name is still `pass_secret_service`. Set the
`meta.mainProgram` attribute to allow for use of `lib.getExe`.

SuperSandro2000 committed:
Merge pull request #211816 from wegank/proto-plus-bump

SuperSandro2000 committed:
Merge pull request #211166 from SuperSandro2000/archisteamfarm

SuperSandro2000 committed:
dnscontrol: 3.23.0 -> 3.24.0

Diff: https://github.com/StackExchange/dnscontrol/compare/v3.23.0...v3.24.0

r-ryantm committed:
godns: 2.9.1 -> 2.9.3

marsam committed:
Merge pull request #211330 from r-ryantm/auto-update/esbuild

esbuild: 0.16.17 -> 0.17.3

ncfavier committed:
nixos: remove stray spaces

ncfavier committed:
Merge pull request #211992 from ncfavier/remove-stray-spaces

figsoda committed:
Merge pull request #211990 from r-ryantm/auto-update/godns

godns: 2.9.1 -> 2.9.3

figsoda committed:
Merge pull request #211989 from SuperSandro2000/dnscontrol

dnscontrol: 3.23.0 -> 3.24.0

jtojnar commented on pkgs/development/libraries/gtk/patches/2.0-sysconfdir_install.patch:
```diff
@@ -0,0 +1,470 @@
+diff --git a/build/win32/vs10/gtk+.sln b/build/win32/vs10/gtk+.sln
+index 57cba1d8b2..8224d0b9a7 100644
+--- a/build/win32/vs10/gtk+.sln
```
Why is this file even being patched?

SuperSandro2000 replied:
Oh, because when closing the git repo the line endings changed.

zhaofengli committed:
texlive: disable luajit on RISC-V (#211999)

Support is not yet there

zowoq committed:
Revert "nixVersions.stable: 2.12 -> 2.13"

This reverts commit 956bc25a803ac00394f2a6d98b7a901ab3065dcf.

broke the tarball job

zowoq committed:
Revert "nixos/nix-fallback-paths: 2.12.0 -> 2.13.1"

This reverts commit 37a44ff3a0415ab84fe52e9c78ad4bafd227737b.

figsoda committed:
Merge pull request #211962 from figsoda/go

treewide: remove empty go vendor hashes

figsoda committed:
Merge pull request #211982 from houstdav000/patch/pass-secret-service-mainProgram

pass-secret-service: Add meta.mainProgram

Et7f3 committed:
pkgs/development/compilers/ocaml: cleanup unused files

Probably left over by #114848

NickCao committed:
Merge pull request #211559 from GTrunSec/nomad-credential

nixos/nomad: add LoadCredential option

wegank committed:
Merge pull request #190041 from r-ryantm/auto-update/python310Packages.flufl_lock

python310Packages.flufl_lock: 7.1 -> 7.1.1

wegank committed:
Merge pull request #189595 from r-ryantm/auto-update/jbang

jbang: 0.97.0 -> 0.102.0

wegank committed:
Merge pull request #212011 from Et7f3/ocaml_cleanup_unused_files

ocaml: cleanup unused files

wegank committed:
Merge pull request #190094 from r-ryantm/auto-update/python310Packages.pyro-ppl

python310Packages.pyro-ppl: 1.8.3 -> 1.8.4

felixzieger committed:
parallel: 20220322 -> 20221122 (#206262)

trofi commented:
103 commits, bad push?
